packages/@icon-magic/distribute/src/index.ts

```diff
@@ -23,28 +22,34 @@ interface ContentImage {
 interface AssetCatalog {
   images?: ContentImage[];
 }
+type ICON_TYPES = 'svg' | 'png' | 'webp' | 'all';
 
 /**
  * Distributes a set of icons to the output folder based on the flag
  * @param iconSet set of icons to be moved to the output folder
- * @param flag createImageSet, distributeByResolution
+ * @param type svg, png, webp, all
  * @param outputPath output directory path to copy the assets to
+ * @param groupByCategory (for sprite creation) whether to group by the category attribute
  * @retuns promise after completion
  */
-export async function distributeByFlag(
+export async function distributeByType(
   iconConfig: IconConfigHash,
   outputPath: string,
-  flag?: string
+  type: ICON_TYPES = 'all',
+  groupByCategory = true
 ): Promise<void> {
-  LOGGER.debug(`entering distribute with ${flag}`);
+  LOGGER.debug(`entering distribute with ${type}`);
   const iconSet = new IconSet(iconConfig, true);
 
+  if (type === 'svg' || type === 'all') {
+    await createSprite(iconSet, outputPath, groupByCategory);
```

Comment:
Should this be handled separately from the rest or should we incorporate it into the default case: `distributeSvg`? In `icon-magic` v1, there is sorting of the icons before the sprite is created, idk if that's something we want to do?